sycl/include/sycl/ext/oneapi/experimental/prefetch.hpp

```diff
@@ -73,10 +73,10 @@ template <typename T, typename Properties>
 void prefetch_impl(T *ptr, size_t bytes, Properties properties) {
 #ifdef __SYCL_DEVICE_ONLY__
   auto *ptrGlobalAS =
-      reinterpret_cast<__attribute__((opencl_global)) const char *>(
+      reinterpret_cast<__attribute__((opencl_global)) const unsigned char *>(
```

Comment:
Note for reviewers: compiler only declares signed and unsigned char overloads, hence the change.

We could probably tweak the compiler to have plain `char` overloads as well, but I'm not sure that it won't cause any unwanted side effects, so I went with this change for now.

`unsigned char` should be a safe choice when it comes to object representation